src/util/resize.js

```diff
@@ -0,0 +1,49 @@
+export const DEFAULT_VERTICAL_FLEX = ['1', '1', '1'];
+export const DIVIDER_HEIGHT = 4;
+export const EDITOR_MIN_HEIGHT = 85;
+
+function isSecondDividerPushingUpFirstDivider(height0, height1, deltaY) {
+  return height0 > EDITOR_MIN_HEIGHT &&
+    height1 === EDITOR_MIN_HEIGHT &&
+    deltaY < 0;
+}
+
+function adjustY(height2, lastY, deltaY) {
+  const maxDeltaY = height2 - EDITOR_MIN_HEIGHT;
+  return lastY + Math.min(deltaY, maxDeltaY);
+}
+
+export function updateVerticalFlex({deltaY, heights, index, lastY, y}) {
+  const [height0, height1, height2] = heights;
+  if (index === 0) {
+    return [
+      `0 1 ${y}px`,
+      '1',
+      height2 ? `0 1 ${height2}px` : '1',
+    ];
+  }
+  const offset = DIVIDER_HEIGHT + height0;
+  const offsetY = y - offset;
```

Comment:
Similarly—`dragTargetDistanceFromTopOfEditor1` (this is not a serious pitch, but something a little more verbose would help a lot)